overlord/snapstate/handlers.go

```diff
@@ -2266,6 +2266,8 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 			t.Logf("reboot postponed to later tasks")
 		}
 	}
+	// XXX: This logic looks a bit confusing, and replacing this simply with
+	// 'rebootInfo.RebootRequired' causes issues. This needs to be looked at
```

Comment:
Yea actually I kinda finally figured the logic out, so this makes more sense now. But it can definitely be reorganized a bit, but i'll leave it out for now.